### ISSUE TITLE
When migrating from 3.2 core to 5.0 preview 7 HttpClient PostAsJsonAsync code stopped working because BaseAddress property seems to be ignored

### ISSUE THREAD

**Issue by RichardVogelij** [dotnet/runtime#39785]
HttpClient's PostAsJsonAsync ignores BaseAddress when passing a Uri - but does not ignore it when passing a string.

I set the BaseAddress property in my Main.cs like so:

```builder.Services.AddSingleton(sp => new HttpClient { BaseAddress = new Uri(builder.HostEnvironment.BaseAddress) });```

My code in a service is doing:

```var response = await _client.PostAsJsonAsync<TokenContainer>(new Uri("/api/User/RefreshToken"), new TokenContainer() { Token = refreshToken });```

which worked fine in .net core 3.2. After updating to .net 5 preview 7 I started to get the following error:

**Only 'http' and 'https' schemes are allowed. (Parameter 'requestUri')**

```
crit: Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer[100]
      Unhandled exception rendering component: Only 'http' and 'https' schemes are allowed. (Parameter 'requestUri')
System.ArgumentException: Only 'http' and 'https' schemes are allowed. (Parameter 'requestUri')
   at System.Net.Http.HttpRequestMessage.InitializeValues(HttpMethod method, Uri requestUri)
   at System.Net.Http.HttpRequestMessage..ctor(HttpMethod method, Uri requestUri)
   at System.Net.Http.HttpClient.CreateRequestMessage(HttpMethod method, Uri uri)
   at System.Net.Http.HttpClient.PostAsync(Uri requestUri, HttpContent content, CancellationToken cancellationToken)
   at System.Net.Http.HttpClientExtensions.PostAsync[TokenContainer](HttpClient client, Uri requestUri, TokenContainer value, MediaTypeFormatter formatter, MediaTypeHeaderValue mediaType, CancellationToken cancellationToken)
   at System.Net.Http.HttpClientExtensions.PostAsync[TokenContainer](HttpClient client, Uri requestUri, TokenContainer value, MediaTypeFormatter formatter, CancellationToken cancellationToken)
   at System.Net.Http.HttpClientExtensions.PostAsJsonAsync[TokenContainer](HttpClient client, Uri requestUri, TokenContainer value, CancellationToken cancellationToken)
   at System.Net.Http.HttpClientExtensions.PostAsJsonAsync[TokenContainer](HttpClient client, Uri requestUri, TokenContainer value)
```

Changing the PostAsJsonAsync to:

```var response = await _client.PostAsJsonAsync<TokenContainer>("/api/User/RefreshToken", new TokenContainer() { Token = refreshToken });``` (note - no new Uri() container for the uri)

seemed to solve this issue. I suppose the behavior should be the same? If this is by design a breaking change note should be added for this.

Thank you for your work on .net 5!

**Comment by mkArtakMSFT** [comment]
@lewing FYI

**Comment by karelz** [comment]
@layomia @Jozkee @eerhardt looks like HttpClientExtensions for Json - can you please check it out if it is bug there?

**Comment by MihaZupan** [comment]
We are not ignoring the BaseAddress. The issue here is that the Uri is created without explicitly specifying `UriKind.Relative` in the constructor.

You should use `new Uri("/api/User/RefreshToken", UriKind.Relative)`, or simply use the string overload of `PostAsJsonAsync`.

`new Uri("/api/User/RefreshToken")` will throw on Windows, but on Linux it will be parsed as an absolute file path.
As it is an absolute path, our validation will throw since the scheme is not correct.

I don't see an overload in AspNetCore's 3.1 implementation taking a `Uri`, so I assume you added the explicit Uri constructor when porting from `PostJsonAsync` to `PostAsJsonAsync`?

Feel free to reopen the issue if this does not solve your problem.

**Comment by Jozkee** [comment]
If you use `new Uri(string)` the instance is created with `UriKind.Absolute`, which is not accepted by `HttpClient` methods if it doesn't start with http or https. When you pass the url using the string overload, the `Uri` is instantiated with `UriKind.RelativeOrAbsolute`.

@RichardVogelij just out of curiosity, what package were you using on .NET core 3.1 (Microsoft.AspNetCore.Blazor.HttpClient or System.Net.Http.Json)? If you were using the latter it would be strange to see a change of behavior between 3.1 and 5.0.

In the case that you were using the former package, as @MihaZupan mentions, there were no overloads taking a `Uri` parameter. How did you manage to pass the `requestUri` parameter as a `Uri` instance?

**Comment by RichardVogelij** [comment]
I'm currently on the road so my hands are somewhat tied to answer you quesrtion, but I can at least mention I was not coming from 3.1 but from 3.2 - I remember having to use Uri type where I used to use a raw string. Also worth mentioning is this code is being used on Blazor Wasm side.

The HttpClient I am using in the "before the update to 5.0" scenario - with the Uri in the call to PostAsJsonAsync<T>() is declared in
Assembly netstandard, Version=2.1.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51

So in essence:

When using this exact type from this .dll in combination with PostAsJsonAsync<T>(Uri) my call worked without specifying the urlkind explicitly. It stopped working when I upgraded my project to .net 5.0 

Assembly System.Net.Http, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
